src/decoders/base65536_decoder.rs

```diff
@@ -0,0 +1,158 @@
+///! Decode a base65536 string
+///! Performs error handling and returns a string
+///! Call base65536_decoder.crack to use. It returns option<String> and check with
+///! `result.is_some()` to see if it returned okay.
+///
+use crate::checkers::CheckerTypes;
+use crate::decoders::interface::check_string_success;
+
+use super::crack_results::CrackResult;
+use super::interface::Crack;
+use super::interface::Decoder;
+
+use log::{debug, info, trace};
+
+/// The base65536 decoder, call:
+/// `let base65536_decoder = Decoder::<Base65536Decoder>::new()` to create a new instance
+/// And then call:
+/// `result = base65536_decoder.crack(input)` to decode a base65536 string
+/// The struct generated by new() comes from interface.rs
+/// ```
+/// use ares::decoders::base65536_decoder::{Base65536Decoder};
+/// use ares::decoders::interface::{Crack, Decoder};
+/// use ares::checkers::{athena::Athena, CheckerTypes, checker_type::{Check, Checker}};
+///
+/// let decode_base65536 = Decoder::<Base65536Decoder>::new();
+/// let athena_checker = Checker::<Athena>::new();
+/// let checker = CheckerTypes::CheckAthena(athena_checker);
+///
+/// let result = decode_base65536.crack("𒅓鹨𖡮𒀠啦ꍢ顡啫𓍱𓁡𠁴唬𓍪鱤啥𖥭𔐠𔕯ᔮ", &checker).unencrypted_text;
+/// assert!(result.is_some());
+/// assert_eq!(result.unwrap(), "Sphinx of black quartz, judge my vow.");
+/// ```
```

Comment:
Do these need to be `///!` ? They do not appear on our docs 🤔  https://docs.rs/project_ares/0.4.0/ares/decoders/base64_decoder/index.html

Reply:
No, I copied it from the Base64 decoder. It's the same.